From 6feec423b50acaa485b4bd555bae93ed13cb1220 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Fri, 21 Sep 2012 18:57:44 -0300
Subject: [RHEL6 qemu-kvm PATCH 23/23] usb-redir: Add chardev open / close
 debug logging

RH-Author: Hans de Goede <hdegoede@redhat.com>
Message-id: <1348253864-3050-23-git-send-email-hdegoede@redhat.com>
Patchwork-id: 42199
O-Subject: [RHEL-6.4 qemu-kvm PATCH 22/22] usb-redir: Add chardev open / close debug logging
Bugzilla: 805172
RH-Acked-by: Uri Lublin <uril@redhat.com>
RH-Acked-by: Arnon Gilboa <agilboa@redhat.com>
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
Upstream commit: 09054d19e72f6991cd359d41c479badd92f62a06
---
 usb-redir.c | 6 ++++++
 1 file changed, 6 insertions(+)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 usb-redir.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/usb-redir.c b/usb-redir.c
index 018dc4f..f2cd610 100644
--- a/usb-redir.c
+++ b/usb-redir.c
@@ -906,6 +906,7 @@ static void usbredir_chardev_close_bh(void *opaque)
     usbredir_device_disconnect(dev);
 
     if (dev->parser) {
+        DPRINTF("destroying usbredirparser\n");
         usbredirparser_destroy(dev->parser);
         dev->parser = NULL;
     }
@@ -920,6 +921,8 @@ static void usbredir_chardev_open(USBRedirDevice *dev)
     usbredir_chardev_close_bh(dev);
     qemu_bh_cancel(dev->chardev_close_bh);
 
+    DPRINTF("creating usbredirparser\n");
+
     dev->parser = usbredirparser_create();
     if (dev->parser == NULL) {
         abort();
@@ -1020,9 +1023,11 @@ static void usbredir_chardev_event(void *opaque, int event)
 
     switch (event) {
     case CHR_EVENT_OPENED:
+        DPRINTF("chardev open\n");
         usbredir_chardev_open(dev);
         break;
     case CHR_EVENT_CLOSED:
+        DPRINTF("chardev close\n");
         qemu_bh_schedule(dev->chardev_close_bh);
         break;
     }
@@ -1288,6 +1293,7 @@ static void usbredir_device_disconnect(void *priv)
     qemu_del_timer(dev->attach_timer);
 
     if (dev->dev.attached) {
+        DPRINTF("detaching device\n");
         usb_device_detach(&dev->dev);
         /*
          * Delay next usb device attach to give the guest a chance to see
-- 
1.7.11.4

